pkg/lib/schema/parser.go

```diff
@@ -0,0 +1,404 @@
+package schema
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/globalsign/mgo/bson"
+
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+)
+
+// JSONSchemaParser parses JSON Schema format into Schema
+type JSONSchemaParser struct{}
+
+// NewJSONSchemaParser creates a new JSON Schema parser
+func NewJSONSchemaParser() Parser {
+	return &JSONSchemaParser{}
+}
+
+// Parse reads a JSON Schema and converts it to Schema
+func (p *JSONSchemaParser) Parse(reader io.Reader) (*Schema, error) {
+	var jsonSchema map[string]interface{}
```

Comment:
maybe use structure for schema? Is it necessary to be able to have custom keys here?

Reply:
maybe for next iteration?